python/ml4ir/applications/classification/pipeline.py

```diff
@@ -0,0 +1,131 @@
+
+from typing import Union, List, Type
+
+
+class ClassificationPipeline(RelevancePipeline):
+    """
+    Pipeline defining the classification models.
+    """
+    def __init__(self, args: Namespace):
+        self.loss_key = args.loss_key
+        super().__init__(args)
+
+    def get_relevance_model(self, feature_layer_keys_to_fns={}) -> RelevanceModel:
+        """
+        Creates RelevanceModel suited for classification use-case.
+
+        NOTE: Override this method to create custom loss, scorer, model objects.
+        """
+
+        # Define interaction model
+        interaction_model: InteractionModel = UnivariateInteractionModel(
+            feature_config=self.feature_config,
+            feature_layer_keys_to_fns=feature_layer_keys_to_fns,
+            tfrecord_type=self.tfrecord_type,
+            file_io=self.file_io)
+
+        # Define loss object from loss key
+        loss: RelevanceLossBase = categorical_cross_entropy.get_loss(
+            loss_key=self.loss_key
+        )
+
+        # Define scorer
+        scorer: ScorerBase = RelevanceScorer.from_model_config_file(
+            model_config_file=self.model_config_file,
+            interaction_model=interaction_model,
+            loss=loss,
+            output_name=self.args.output_name,
+            logger=self.logger,
+            file_io=self.file_io,
+        )
+
+        # Define metrics objects from metrics keys
+        metrics: List[Union[Type[Metric], str]] = [
+            metric_factory.get_metric(metric_key=metric_key) for metric_key in self.metrics_keys
+        ]
+
+        # Define optimizer
+        optimizer: Optimizer = get_optimizer(
+            optimizer_key=self.optimizer_key,
+            learning_rate=self.args.learning_rate,
+            learning_rate_decay=self.args.learning_rate_decay,
+            learning_rate_decay_steps=self.args.learning_rate_decay_steps,
+            gradient_clip_value=self.args.gradient_clip_value,
+        )
+
+        # Combine the above to define a RelevanceModel
+        relevance_model: RelevanceModel = RelevanceModel(
+            feature_config=self.feature_config,
+            scorer=scorer,
+            metrics=metrics,
+            optimizer=optimizer,
+            tfrecord_type=self.tfrecord_type,
+            model_file=self.args.model_file,
+            compile_keras_model=self.args.compile_keras_model,
+            output_name=self.args.output_name,
+            file_io=self.local_io,
+            logger=self.logger,
+        )
+        return relevance_model
+
+    def get_relevance_dataset(self, preprocessing_keys_to_fns={}) -> RelevanceDataset:
+        """
+        Creates RelevanceDataset
+
+        NOTE: Override this method to create custom dataset objects
+        """
+
+        # Adding one_hot_vectorizer needed for classification.
+        preprocessing_keys_to_fns = {
+            "one_hot_vectorize_label": get_one_hot_vectorizer(self.feature_config.get_label(), self.file_io)
```

Comment:
@Ullimague You shouldn't be doing this. Add it to `PreprocessingMap` instead and use it from feature config YAML.

Reply:
If you want to pass in `file_io`, then you can change the function definition of the preprocessing functions to take in `file_io` or `**kwargs`

Reply:
Implemented, this makes only sense when a label is a categorical feature, hence only for classification thought..

Reply:
@Ullimague and I discussed offline. Adding this to PreprocessingMap causes issues downstream while saving the model. So we decided to keep this as on-demand that way we can make sure all functions in PreprocessingMap are save-able. @Ullimague will be reverting this to its original implementation and using it from `pipeline.py`. But make sure you file an issue for fixing `one_hot_vectorizer` so that it does not need to be wrapped as a Keras model. That's the real issue here.